File: frontend/types/briefing.ts
export interface AffectedRegion {
  region: string;
  alert_count: number;
  max_severity: string;
  alert_ids: string[];
}

export interface SourceCount {
  source: string;
  label: string;
  count: number;
}

export interface CountryCount {
  code: string;
  count: number;
}

export interface MajorEvent {
  title: string;
  severity: string;
  source: string;
  category?: string;
  region?: string;
  alert_id: string;
}

export interface CrossBorderPattern {
  type: string;
  description: string;
  alert_ids: string[];
  confidence: string;
}

export interface TrendAnomaly {
  dimension: string;
  key: string;
  current_count: number;
  rolling_avg: number;
  ratio: number;
}

export interface PotentialImplications {
  economy: string[];
  logistics: string[];
  infrastructure: string[];
  technology: string[];
  finance: string[];
}

export interface ObservedEventItem {
  canonical_event_id: string;
  event_title: string;
  event_type?: string;
  severity: string;
  observed_count: number;
  sources: string[];
  source_ids: string[];
}

export interface ObservedEventsSection {
  summary: string;
  items: ObservedEventItem[];
  confidence: string;
}

export interface VerifiedExposureItem {
  event_id: string;
  event_title: string;
  asset_name: string;
  asset_type: string;
  exposure_type: string;
  confidence: string;
  source_ids: string[];
}

export interface VerifiedExposureSection {
  summary: string;
  items: VerifiedExposureItem[];
  confidence: string;
}

export interface SourcedClaim {
  description: string;
  confidence: string;
  evidence_level?: string;
  source_ids: string[];
}

export interface SectionConfidence {
  observed_events: string;
  verified_exposure: string;
  potential_implications: string;
  confirmed_impacts: string;
  cross_border_relevance: string;
  technology_infrastructure_risks: string;
}

export interface BriefingContent {
  generated_at: string;
  type: "rule_based" | "llm";
  active_count: number;
  overall_risk_score: number;
  summary: string;
  overall_confidence: "low" | "medium" | "high";
  by_source: SourceCount[];
  top_countries: CountryCount[];
  affected_regions: AffectedRegion[];
  major_events: MajorEvent[];
  cross_border_patterns: CrossBorderPattern[];
  trend_anomalies: TrendAnomaly[];
  potential_implications: PotentialImplications;
  observed_events?: ObservedEventsSection;
  verified_exposure?: VerifiedExposureSection;
  confirmed_impacts?: SourcedClaim[];
  cross_border_relevance?: SourcedClaim[];
  technology_infrastructure_risks?: SourcedClaim[];
  evidence_gaps?: string[];
  section_confidence?: SectionConfidence;
  limitations: string[];
  source_alert_ids: string[];
  score_breakdown: Record<string, unknown>;
}

export interface Briefing {
  id: string;
  generated_at: string;
  type: "rule_based" | "llm";
  overall_risk_score: number;
  overall_confidence: "low" | "medium" | "high";
  content: BriefingContent;
  source_alert_ids: string[];
  llm_model: string | null;
}

export interface BriefingListResponse {
  items: Briefing[];
  total: number;
  limit: number;
  offset: number;
}

File: backend/app/schemas/briefing.py
"""Briefing API schemas."""

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class AffectedRegion(BaseModel):
    region: str
    alert_count: int
    max_severity: str
    alert_ids: list[str] = Field(default_factory=list)


class SourceCount(BaseModel):
    source: str
    label: str
    count: int


class CountryCount(BaseModel):
    code: str
    count: int


class MajorEvent(BaseModel):
    title: str
    severity: str
    source: str
    category: str | None = None
    region: str | None = None
    alert_id: str


class CrossBorderPattern(BaseModel):
    type: str
    description: str
    alert_ids: list[str] = Field(default_factory=list)
    confidence: str = "low"


class TrendAnomalyItem(BaseModel):
    dimension: str
    key: str
    current_count: int
    rolling_avg: float
    ratio: float


class PotentialImplications(BaseModel):
    economy: list[str] = Field(default_factory=list)
    logistics: list[str] = Field(default_factory=list)
    infrastructure: list[str] = Field(default_factory=list)
    technology: list[str] = Field(default_factory=list)
    finance: list[str] = Field(default_factory=list)


class BriefingContent(BaseModel):
    generated_at: str
    type: str = "rule_based"
    active_count: int = 0
    overall_risk_score: int = 0
    summary: str = ""
    overall_confidence: str = "low"
    by_source: list[SourceCount] = Field(default_factory=list)
    top_countries: list[CountryCount] = Field(default_factory=list)
    affected_regions: list[AffectedRegion] = Field(default_factory=list)
    major_events: list[MajorEvent] = Field(default_factory=list)
    cross_border_patterns: list[CrossBorderPattern] = Field(default_factory=list)
    trend_anomalies: list[TrendAnomalyItem] = Field(default_factory=list)
    potential_implications: PotentialImplications = Field(default_factory=PotentialImplications)
    limitations: list[str] = Field(default_factory=list)
    source_alert_ids: list[str] = Field(default_factory=list)
    score_breakdown: dict[str, Any] = Field(default_factory=dict)


class BriefingResponse(BaseModel):
    id: UUID
    generated_at: datetime
    type: str
    overall_risk_score: int
    overall_confidence: str
    content: BriefingContent
    source_alert_ids: list[UUID] = Field(default_factory=list)
    llm_model: str | None = None


class BriefingListResponse(BaseModel):
    items: list[BriefingResponse]
    total: int
    limit: int
    offset: int


class GenerateBriefingRequest(BaseModel):
    type: str = "auto"


class GenerateBriefingResponse(BaseModel):
    briefing_id: str
    type: str
    overall_risk_score: int
    generated_at: datetime
